src/core/backbone_router/leader.cpp

```diff
@@ -186,6 +235,64 @@ void Leader::Update(void)
 #endif
 }
 
+void Leader::UpdateDomainPrefixConfig(void)
+{
+    NetworkData::Iterator           iterator = NetworkData::kIteratorInit;
+    NetworkData::OnMeshPrefixConfig config;
+    DomainPrefixState               state;
+    bool                            found = false;
+
+    memset(&config, 0, sizeof(config));
+
+    while (Get<NetworkData::Leader>().GetNextOnMeshPrefix(iterator, config) == OT_ERROR_NONE)
+    {
+        if (config.mDp)
+        {
+            found = true;
+            break;
+        }
+    }
+
+    if (!found)
+    {
+        if (mDomainPrefix.mLength != 0)
+        {
+            // Domain Prefix does not exist any more.
+            memset(&mDomainPrefix, 0, sizeof(mDomainPrefix));
+            state = kDomainPrefixRemoved;
+        }
+        else
+        {
+            state = kDomainPrefixNone;
+        }
+    }
+    else if (config.mPrefix.mLength == mDomainPrefix.mLength &&
+             otIp6PrefixMatch(&mDomainPrefix.mPrefix, &config.mPrefix.mPrefix) >= mDomainPrefix.mLength)
+    {
+        state = kDomainPrefixUnchanged;
+    }
+    else
+    {
+        mDomainPrefix = config.mPrefix;
+
+        if (mDomainPrefix.mLength == 0)
```

Comment:
This is effectively the same as `if (config.mPrefix == 0)`. Is this what was intended?

Reply:
Good catch!